<a href="https://colab.research.google.com/github/Haris-09/Python-Practice/blob/main/Python_Primer.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#The order of Operations

In [ ]:
5 + 30 * 20

605

For example, in the following equation, the numbers 30 and 20 are multiplied first, and the number 5 is added to their product.

Parentheses can change the order of operations.

This equation is another way of saying, “multiply 30 by 20, and then add 5 to the result.” The result is 605. We can change the order of operations by adding parentheses around the first two numbers, like so:

In [ ]:
(5 + 30) * 20

700

The result of this equation is 700 (not 605) because the parentheses tell Python to do the operation in the parentheses first, and then do the operation outside the parentheses.

#Nested Parentheses
Inner and outer Parentheses

In [ ]:
((5+30)*20)/10

70.0

In this case, Python evaluates the innermost parentheses first, then the outer ones, and then the final division operator. In other words, this equation is saying, “add 5 to 30, then multiply the result by 20, and divide that result by 10.” Here’s what happens:
• Adding 5 to 30 gives 35.
• Multiplying 35 by 20 gives 700.
• Dividing 700 by 10 gives the final answer of 70.

#Variables
Variable in programming means a place to store information such as numbers, text, list of numbers and text etc.Its like variable is a label for something.To create a variable name Ali, we use an equal sign =

In [ ]:
Ali=100

Variable names can be made up of letters, numbers and the underscore character ( _ ), but they cant start with a number. You can use anything from single letters such as (a) to long sentences for variable names. But you cannot use space ( )A short name variable is best 

#Variable exercise

In [ ]:
ali_coins=20
fahad_coins=10
abdullah_coins=10

ali_coins+fahad_coins+abdullah_coins

40

#Strings in Python

we create a string by putting quotes around text. one can also use sigle or double quotes

In [3]:
abubakr = 'first caliph!!'
print(abubakr)

first caliph!!


In [4]:
abubakr = "first caliph!!"
print(abubakr)

first caliph!!


#Syntax and Syntax Error

*   Syntax means the arrangement and order of word in a sentence.
*   Syntax error means that you did something in an order python was not expecting or you missed something.


In [2]:
abubakr = 'first caliph!!
print(abubakr)

SyntaxError: ignored

#Strings Problems handling

**Adding second line to your command**


*   You can use 3 single quotes to write a second or third or as many lines as you want to a given command.



In [5]:
abubakr= '''first
Righteous
caliph!!'''
print(abubakr)

first
Righteous
caliph!!


to handle a sentence which contains multiple single or double quotes use 3 single quotes to handle this problem.

In [6]:
silly_string= '''He said, "Aren't can't shouldn't wouldn't."'''
print(silly_string)

He said, "Aren't can't shouldn't wouldn't."


**Adding a backslash  \or escaping**


* \or excapingmeans “Yes, I know I have quotes inside my string, and I want you to ignore them until you see the end quote.

In [7]:
single_quote_str= 'He said, "Aren\'t can\'t shouldn\'t wouldn\'t."'
print (single_quote_str)

He said, "Aren't can't shouldn't wouldn't."


In [8]:
double_quote_str= "He said, \"Aren't can't shouldn't wouldn't.\""
print (double_quote_str)

He said, "Aren't can't shouldn't wouldn't."
